src/lcm/max_Q_over_c.py

```diff
@@ -0,0 +1,112 @@
+import functools
+from collections.abc import Callable
+
+import jax.numpy as jnp
+from jax import Array
+
+from lcm.argmax import argmax_and_max
+from lcm.dispatchers import productmap
+from lcm.typing import ArgmaxQOverCFunction, MaxQOverCFunction, ParamsDict, Scalar
+
+
+def get_max_Q_over_c(
+    Q_and_F: Callable[..., tuple[Array, Array]],
+    continuous_actions_names: tuple[str, ...],
+    states_and_discrete_actions_names: tuple[str, ...],
+) -> MaxQOverCFunction:
+    r"""Get the function returning the maximum of Q over continuous actions.
+
+    The state-action value function $Q$ is defined as:
+
+    ```{math}
+    Q(x, a) =  g(U(x, a), \mathbb{E}[V(x', a') | x, a]),
```

Comment:
```suggestion
    Q(x, a) =  H(U(x, a), \mathbb{E}[V(x', a') | x, a]),
```

My bad. I think `H` is sometimes used for this aggregator. And we should stick to uppercase, I suppose. Affects the line below, too (and the paper).